### ISSUE TITLE
npm install fails on android ubuntu because of ENOSYS

### ISSUE THREAD

**Issue by martindobias** [npm/npm#13665]
#### What's going wrong?
I have installed successfully nodejs on android debian release using GNURoot (https://play.google.com/store/apps/details?id=com.gnuroot.debian&hl=en)
Everything works fine until I try to install some npm package

#### How can the CLI team reproduce the problem?

```console
root@localhost:~# npm install -g elm
npm ERR! Linux 3.10.61-7884513
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install" "-g" "elm"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.3
npm ERR! path /tmp/npm-17037-bd25ae3c/registry.npmjs.org/elm/-
npm ERR! code ENOSYS
npm ERR! errno -38
npm ERR! syscall mkdir

npm ERR! ENOSYS: function not implemented, mkdir '/tmp/npm-17037-bd25ae3c/registry.npmjs.org/elm/-'
```

#### Possible cause
It seems the problem is when calling
`mkdir '/tmp/npm-17037-bd25ae3c/registry.npmjs.org/elm/-'`
Exactly same error ("function not implemented") is returned when this command is run manually

As it is a linux distro the command should be
`mkdir -p '/tmp/npm-17037-bd25ae3c/registry.npmjs.org/elm/-'`

Could this be the problem?
Am I able to force npm to work correctly?

Thanks a lot

**Comment by RedBird1** [comment]
Hey any updates on this? Currently trying to run npm install using termux on android, and I'm having the same errors.

**Comment by othiym23** [comment]
```
npm ERR! ENOSYS: function not implemented, mkdir '/tmp/npm-17037-bd25ae3c/registry.npmjs.org/elm/-'
```

npm needs to be able to create new directories. This is a failing system call within Node itself, and it means that either the current filesystem or `$TMPDIR` don't support making new directories. You can try to fix this by either setting `TMPDIR` in your termux session on Android, or by [configuring the `tmp` value](https://docs.npmjs.com/misc/config#tmp) within npm. That said, if that's not enough to get it to work, I'm not sure there's a lot the npm team can do. Android is sufficiently different from vanilla Linux (even within a Debian chroot) that supporting it is tricky, and Android is _definitely_ not on the list of platforms supported by the CLI team. Good luck, and thanks for your time!

**Comment by martindobias** [comment]
why are you closing this?
i have shown that the problem is not with directories. from command line the dirs can be created nicely.
the problem is that the whole chain of dirs is being created at once and not directory by directory
please change the functionality so it is environment sensitive and provides the flags for successful chain dir creation
and do not close issues without reading them first
thanks
